QImportP.wsf

```diff
@@ -103,230 +100,551 @@ var QEXE_DEFAULT = Array(
+        };
+		
+    this.toQuicken = function (histDate, histPrice, symb)
+        { 
+		// output is symb,date,price
+		
+		// symb leading with ! means ignore "/div" option
+		var dodiv = (DIV_NUM > 0);
+		if (symb.charAt(0) == "!")
+			{
+			dodiv = false;
+		    symb = symb.substr(1);
+			}
+			
+		if (dodiv)
+		    {
+			histPrice = Math.round(parseFloat(histPrice)/DIV_NUM * 10000)/10000;
+			}
+			
+		// if SYMONLY, want to output symbol excluding market
+		if (SYMONLY)
+			 symb = symb.substr(symb.indexOf(":")+1);
+
+		
+		// DATE_FORMAT = parameter or default D/M/Y (eg. D/M/Y or M-D-Y) need to replace D with Day, M with Month, Y with year
+		var myDateOut = DATE_FORMAT;
+		var myDate = new Date(histDate);
+		myDateOut = myDateOut.replace("D",trunc2(myDate.getDate()));
+		myDateOut = myDateOut.replace("M",trunc2(myDate.getMonth()+1));
+		myDateOut = myDateOut.replace("Y",myDate.getFullYear());
+		return symb + ',' + myDateOut + ',' + histPrice;
+					
+		};
+		
+	this.getPriceDate = function (ResponseText,AllPrices,AllDates)
+		{
+		// setup domain parsor to process xml data
+		var myDom = new ActiveXObject("MSXML2.DOMDocument.3.0");
+		myDom.async=false;
+		myDom.resolveExternals=false;
+		
+		//var AllPrices = [];
+		//var AllDates = [];
+		 
+		 try  // regex may return error if no matches so catch any to avoid script abort
+			{
+			var webHtml = ResponseText.replace(/(?:\r\n|\r|\n|\\n|")/g, "");  // replace lf, cr, "\n" and "(quote) with nothing
+			
+			// First get the google provided symbol name from the "meta Description block" because user data may not have the market prefix
+	        //  and there is specific action if the symbol is for a mutual fund (MUTF (including MUTF_GB etc))
+			var descrBlock = webHtml.match(/<meta name=Description.+?>/);   // market:symbol is contained within meta description
+			if (descrBlock == null)
+				{
+				message("Warn: No descrBlock returned",WARNING_LEVEL);
+				return false;
+				}
+			descrBlock = descrBlock[0];
+			
+			var webMktSymb = descrBlock.match(/\(.+\)/);  // market:sybmol is surrounded by (....)
+			// if nothing found in brackets, chances are user supplied invalid symbol
+			if (webMktSymb == null)
+                {
+				message("Warn: Symbol not found in Google history",WARNING_LEVEL);
+				return false;
                 }
-            }
-        }
-        return ret;
-    };
-    this.toString = function () { // just here for debugging
-        return this.url;
-    };
-}
+				
+			webMktSymb = webMktSymb[0];
+			var webMktSymb = webMktSymb.replace(/\(|\)/g,"");  // remove the brackets
+			var webMarket = webMktSymb.substr(0,webMktSymb.indexOf(":"));  // part before the ":"
+			var webSymb = webMktSymb.substr(webMktSymb.indexOf(":")+1);     // part after the ":"
+
+			// Never used because history url returns invalid symbol
+            // if webMarket = CURRENCY, there is no history 
+            //if (webMarket == "CURRENCY")

```

Comment:
No, it is where I was catering for CURRENCY type having no history ... but Google doesn't return any values so the previous chack rejects and this code would never be executed for a CURRENCY type.
Comments now removed.